pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/channel/ServiceUnitStateChannelImpl.java

```diff
@@ -804,7 +804,9 @@ private void handleReleaseEvent(String serviceUnit, ServiceUnitStateData data) {
             if (isTransferCommand(data)) {
                 next = new ServiceUnitStateData(
                         Assigning, data.dstBroker(), data.sourceBroker(), getNextVersionId(data));
-                unloadFuture = closeServiceUnit(serviceUnit, false);
+                // If the optimized bundle unload is disabled, disconnect the clients at time of RELEASE.
+                var disconnectClients = !pulsar.getConfig().isLoadBalancerOptimizeBundleUnload();
```

Comment:
Don't we need to skip `closeServiceUnit ` at `handleOwnEvent` when isLoadBalancerOptimizeBundleUnload = false?

Reply:
yeah. I agree making this as a non-dynamic is safer to handle such cases.